src/main/webapp/resources/js/pages/analysis/components/AnalysisSamples.jsx

```diff
@@ -0,0 +1,118 @@
+import { getI18N } from "../../../utilities/i18n-utilties";
+
+import {
+    getAnalysisInputFiles,
+    getForwardFile,
+    getReverseFile
+} from "../../../apis/analysis/analysis";
+
+export default function AnalysisSamples() {
+  const { analysisDetailsContext, loadAnalysisSamples } = useContext(AnalysisDetailsContext);
+
+  useEffect(() => {
+    loadAnalysisSamples();
+  }, []);
+
+  const downloadReferenceFile = () => {
+    if (analysisDetailsContext.referenceFile.identifier !== undefined)
+    {
+        window.open(`${window.TL.BASE_URL}referenceFiles/download/${analysisDetailsContext.referenceFile.identifier}`, '_blank');
+    }
+  };
+
+  const renderReferenceFile = () => {
+    const referenceFile = [];
+
+    if(analysisDetailsContext.referenceFile.length == 0)
+    {
+        return null;
+    }
+    else {
+
+        referenceFile.push(
+            <h4 key="reference-file-0-heading" style={{fontWeight: "bold"}}>{getI18N("analysis.tab.content.samples.reference-file")}</h4>,
+            <Row key="row-filename">
+                <span key="reference-file-0-file">{analysisDetailsContext.referenceFile.label}</span>
+                <Button
+                    key="reference-file-0-download-button"
+                    className="pull-right"
+                    style={{marginTop: "0.5em"}} type="primary"
+                    onClick={() => {downloadReferenceFile()}}
+                >
+                    <Icon type="download" /> {getI18N("analysis.tab.content.samples.download-reference-file")}
+                </Button>
+            </Row>,
+         );
+         return referenceFile;
+     }
+  };
+
+  const renderSamples = () => {
+    const samplesList = [];
+
+    if(analysisDetailsContext.sequenceFilePairList.length > 0) {
+        let pairIndex = 0;
+        for (let i = 0; i < analysisDetailsContext.samples.length; i++) {
+          samplesList.push(
+            <Card
+                type="inner"
+                title={
+                    <a
+                        target="_blank"
+                        href={`${window.TL.BASE_URL}samples/${analysisDetailsContext.samples[i].sample.identifier}/details`}>
+                            <Icon type="filter" rotate="180" /> {analysisDetailsContext.samples[i].sample.sampleName}
+                    </a>}
+                key={`sample${i}`}
+            >
+                <Row>
+                    <a
+                        target="_blank"
+                        href={`${window.TL.BASE_URL}sequenceFiles/${analysisDetailsContext.samples[i].sequenceFilePair.identifier}/file/${analysisDetailsContext.sequenceFilePairList[pairIndex].identifier}/summary`}>
+                            <Icon type="arrow-right" /> {analysisDetailsContext.sequenceFilePairList[pairIndex].label}
+                   </a>
+                   <span style={{float: 'right'}}>{analysisDetailsContext.sequenceFileSizeList[pairIndex]}</span>
```

Comment:
Really try to avoid floats if you can, they can get tricky, especially because I see you are not clearing the float here.  Look again at [`flexbox`](https://css-tricks.com/snippets/css/a-guide-to-flexbox/) for this.

Reply:
Fixed in 239318b0bb57f7ffc2020d24cf154c055cf22b72